Public/Src/Cache/DistributedCache.Host/Configuration/DistributedContentSettings.cs

```diff
@@ -178,6 +180,24 @@ private DistributedContentSettings(IDictionary<string, string> connectionSecretN
         /// </summary>
         [DataMember]
         public bool UseCompressionForCopies { get; set; } = false;
+
+        /// <summary>
+        /// Upper bound on number of cached GRPC clients.
+        /// </summary>
+        [DataMember]
+        public int MaxGrpcClientCount { get; set; } = DefaultMaxConcurrentCopyOperations;
+
+        /// <summary>
+        /// Maximum cached age for GRPC clients.
+        /// </summary>
+        [DataMember]
+        public int MaxGrpcClientAgeMinutes { get; set; } = 55;
+
+        /// <summary>
+        /// Time between GRPC cache cleanups.
+        /// </summary>
+        [DataMember]
+        public int GrpcClientCleanupDelayMinutes { get; set; } = 17;

```

Comment:
Why 17:)

Reply:
Random starting point that is not a factor of the maximum client age. This should change in the future.

---
In reply to: [290519019](https://github.com/microsoft/BuildXL/pull/391#discussion_r290519019) [](ancestors = 290519019)